tests/cupyx_tests/scipy_tests/special_tests/test_basic.py

```diff
@@ -150,6 +150,12 @@ def test_expm1(self, xp, scp, dtype):
             vals = vals[:, xp.newaxis] + 1j * vals[xp.newaxis, :]
         return scp.special.expm1(vals)
 
+    @testing.for_dtypes("efd")
+    @numpy_cupy_allclose(scipy_name="scp", rtol=1e-6)
+    def test_cosm1(self, xp, scp, dtype):
+        vals = xp.linspace(-50, 50, 200, dtype=dtype)
```

Comment:
Can we have a test with inputs near zero? 
```py
>>> import numpy as np
>>> import scipy.special as sc
>>> x = np.linspace(-1e-8, 1e-8, 10)
>>> np.cos(x)-1
array([-1.11022302e-16,  0.00000000e+00,  0.00000000e+00,  0.00000000e+00,
        0.00000000e+00,  0.00000000e+00,  0.00000000e+00,  0.00000000e+00,
        0.00000000e+00, -1.11022302e-16])
>>> sp.cosm1(x)
array([-5.00000000e-17, -3.02469136e-17, -1.54320988e-17, -5.55555556e-18,
       -6.17283951e-19, -6.17283951e-19, -5.55555556e-18, -1.54320988e-17,
       -3.02469136e-17, -5.00000000e-17])
```

Reply:
I've added a new test for this in the latest commit. Just for reference as well this is the result on the added function, which is the same as scipy:

```python
>>> import cupyx.scipy.special as scp
>>> import cupy as cp
>>> x = cp.linspace(-1e-8, 1e-8, 10)
>>> scp.cosm1(x)
array([-5.00000000e-17, -3.02469136e-17, -1.54320988e-17, -5.55555556e-18,
       -6.17283951e-19, -6.17283951e-19, -5.55555556e-18, -1.54320988e-17,
       -3.02469136e-17, -5.00000000e-17])
```

Reply:
Thanks!